app/assets/src/components/common/DetailsSidebar/PipelineStepDetailsMode/PipelineStepDetailsMode.jsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import cs from "./pipeline_step_details_mode.scss";
+
+class PipelineStepDetailsMode extends React.Component {
+  renderInputFiles() {
+    const { inputFiles } = this.props;
+    if (inputFiles && inputFiles.length) {
+      const fromStepNameToFile = {};
+      inputFiles.forEach(file => {
+        if (!(file.fromStepName in fromStepNameToFile)) {
+          fromStepNameToFile[file.fromStepName] = [];
+        }
+        fromStepNameToFile[file.fromStepName].push(file);
+      });
+
+      const fileGroupList = Object.keys(fromStepNameToFile).map(
+        fromStepName => {
+          const fileGroup = fromStepNameToFile[fromStepName];
+          const fileList = fileGroup.map((file, i) => {
+            return (
+              <div className={cs.link} key={file.fileName + i.toString()}>
```

Comment:
we usually just use string interpolation (which would also save you the toString) like `${file.fileName}-${i}`